src/Val.h

```diff
@@ -1365,6 +1414,16 @@ class VectorVal final : public Val, public notifier::detail::Modifiable {
 	 */
 	const ValPtr& At(unsigned int index) const;
 
+	/**
+	 * Returns the given element treated as a Count type, to efficiently
+	 * support a common type of vector access if we change the underlying
+	 * vector representation.
+	 * @param index  The position in the vector of the element to return.
+	 * @return  The element's value, as a Count underlying representation.
+	 */
+	bro_uint_t CountAt(unsigned int index) const
```

Comment:
Not opposed to this, but curious: why does this special-case deserve its own method?

Reply:
A common idiom for using `AsVector()` is to iterate through its elements, accessing them `AsCount()`.  By adding this method, we can support the idiom and if/when we change the internal representation of vectors those accesses become much more efficient, as they don't require an intermediary `Val` to be constructed.  So similar to the `GetXYZField()` methods for records, but so far not needing that generality since there aren't other idioms in the code base.

Reply:
(and on more instance of `why AsX?` I believe :-)